src/Microsoft.OpenApi.OData.Reader/Common/Utils.cs

```diff
@@ -129,6 +133,74 @@ internal static string NavigationPropertyPath(this ODataPath path, string naviga
                 || s is ODataStreamContentSegment || s is ODataStreamPropertySegment)).Select(e => e.Identifier));
 
             return navigationPropertyName == null ? value : $"{value}/{navigationPropertyName}";
+        }                
+
+        /// <summary>
+        /// Adds a mapping of custom extension values against custom attribute values for a given element to the provided
+        /// extensions object.
+        /// </summary>
+        /// <param name="extensions">The target extensions object in which the mapped extensions and custom attribute
+        /// values will be added to.</param>
+        /// <param name="context">The OData context.</param>
+        /// <param name="element">The target element.</param>
+        internal static void AddCustomAtributesToExtensions(this IDictionary<string, IOpenApiExtension> extensions, ODataContext context, IEdmElement element)
+        {
+            if (extensions  == null ||
+                context == null ||
+                element == null)
+            {
+                return;
+            }
+
+            Dictionary<string, string> atrributesValueMap = GetCustomXMLAtrributesValueMapping(context.Model, element, context.Settings.CustomXMLAttributesMapping);
+
+            if (atrributesValueMap?.Any() ?? false)
+            {
+                foreach (var item in atrributesValueMap)
+                {
+                    if (!extensions.ContainsKey(item.Key))
+                    {
+                        extensions.Add(item.Key, new OpenApiString(item.Value));
```

Comment:
```suggestion
                        extensions.TryAdd(item.Key, new OpenApiString(item.Value));
```
and remove the surrounding if block

Reply:
Unfortunately, `IDictionary<string, IOpenApiExtension>` does not contain a definition for `TryAdd`

Reply:
Unfortunately, `IDictionary<string, IOpenApiExtension>` does not contain a definition for `TryAdd`

Reply:
It's an extension method https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.collectionextensions.tryadd?view=net-6.0

Reply:
This is applicable to the [below versions](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.collectionextensions.tryadd?view=net-6.0#applies-to) of the .NET and .NET Standard:

![image](https://user-images.githubusercontent.com/40403681/166837068-af30bde2-2259-41d1-9973-662d85a06703.png)

The current project is on [.NET Standard 2.0](https://github.com/microsoft/OpenAPI.NET.OData/blob/429e664a27d8ff074d2866eaa5979b0ac142f07f/src/Microsoft.OpenApi.OData.Reader/Microsoft.OpenAPI.OData.Reader.csproj#L15) which doesn't support that extension method:

![image](https://user-images.githubusercontent.com/40403681/166837900-aa9af115-35d4-4bb3-83d1-3d05874ad3a2.png)

Reply:
right, ideally then we'd define our own extension method, this way we avoid duplication, and the day this lib upgrades the target, it'll be easier to identify the places to replace it.

Reply:
Resolved.